### ISSUE TITLE
ScoreCAM paper Algorithm1 implementation question

### ISSUE THREAD

**Issue by CHILLQQ** [haofanwang/Score-CAM#31]
Hello,

I have a question about the implementation of the Algorithm1 of the ScoreCAM paper.
The code
```
              # how much increase if keeping the highlighted region
              # predication on masked input
              output = self.model_arch(input * norm_saliency_map)
              output = F.softmax(output)
              score = output[0][predicted_class]
```
suggests that the output is simply the masked images run through the original neural net. However, in the paper there is an additional step:
$S^{c} = f^c(M) - f^c(X_b)$.

I am not sure exactly why this step is needed in the first place, but since it is in the paper, I am curious why it does not seem to be in the code?

Thank you.

**Comment by usamahassan965** [comment]
Just comparing with baseline . But if you see the methodology diagram , this step is not done there.
